### PR TITLE
[entropy] address audit: 3. safe owner transfer

### PR DESCRIPTION
**Pull request by 0xfirefist**

### CODE REVIEW

**Review by jayantk — COMMENTED** [review]
This generally makes sense. I mostly left nitpicky code comments on naming and documentation. I realize these seem like minor details, but it does affect readability and understandability of the code.

Main thing to change here though is to keep the ownership parameter in the constructor.

File: target_chains/ethereum/contracts/contracts/entropy/EntropyState.sol [hunk 1]
```diff
@@ -34,6 +34,9 @@ contract EntropyInternalStructs {
         mapping(bytes32 => EntropyStructs.Request) requestsOverflow;
         // Mapping from randomness providers to information about each them.
         mapping(address => EntropyStructs.ProviderInfo) providers;
+        // Account address to which the current owner raise a transfer request.
+        // This account will have to claim the ownership to become a new owner.
+        address ownershipTransferAccount;
```
[the remark this answers is not in the document]
Comment (0xfirefist) [reply target_chains/ethereum/contracts/contracts/entropy/EntropyState.sol:39]: I think next owner works well

**Comment by 0xfirefist** [comment]
I was struggling coming up with the right names. I appreciate these suggestions. 
- method - requestOwnershipTransfer(), event - OwnershipTransferRequested
- method - acceptOwnershipTransfer(), event - AcceptOwnershipTransfer
- storing - nextOwner

File: target_chains/ethereum/contracts/contracts/entropy/EntropyUpgradable.sol [hunk 2]
```diff
@@ -40,9 +42,6 @@ contract EntropyUpgradable is
             defaultProvider,
             prefillRequestStorage
         );
-
-        // We need to transfer the ownership from deployer to the new owner
-        transferOwnership(owner);
```
[the remark this answers is not in the document]
Comment (0xfirefist) [reply target_chains/ethereum/contracts/contracts/entropy/EntropyUpgradable.sol:45]: re introduced

**Review by jayantk — APPROVED** [review]

**Comment by 0xfirefist** [comment]
@jayantk, not merging this. Turns out we have this thing already implemented by openzepplin.

**Pull request closed without merging**